contracts/src/Shadowlings.sol

```diff
@@ -10,86 +10,86 @@ import {Verifier as RecoveryVerifier} from "./verifiers/recovery/Verifier.sol";
 import {Verifier as RegisterVerifier} from "./verifiers/register/Verifier.sol";
 
 contract Shadowlings is IAccount, Verifier {
+    bytes32 private constant _R = bytes32(uint256(keccak256("Shadowlings.r")) - 1);
     bytes32 public constant DOMAIN_TYPEHASH = keccak256("EIP712Domain(uint256 chainId,address verifyingContract)");
     bytes32 public constant TRANSFER_TYPEHASH = keccak256("Transfer(address token,address to,uint256 amount)");
 
-    bytes public constant SIGNATURE =
-        hex"0174a52317b658076e35432533edc88c2f86823e2fcfd2b56f8fad46fb32d6a51718811e130eeacc4232614ef16382b62d0d6e04eadf9fb575647e9cca12f0147f";
-
+    address private immutable _SELF;
     address public immutable ENTRY_POINT;
     RecoveryVerifier public immutable RECOVERY;
     RegisterVerifier public immutable REGISTER;
 
     mapping(uint256 => bool) public nullified;
 
+    error NotDelegated();
     error UnsupportedEntryPoint();
     error UnsupportedCall();
-    error OnlyWithoutPrefund();
     error Nullified();
+    error Unauthorized();
     error InvalidProof();
+    error TransferFailed();
 
-    event RecoverySaltHash(address indexed shadowling, uint256 saltHash);
+    event RecoverySaltHash(uint256 saltHash);
 
     constructor(address entryPoint) {
+        _SELF = address(this);
         ENTRY_POINT = entryPoint;
         RECOVERY = new RecoveryVerifier();
         REGISTER = new RegisterVerifier();
     }
 
-    modifier onlyEntryPoint() {
-        if (msg.sender != ENTRY_POINT) {
-            revert UnsupportedEntryPoint();
-        }
+    modifier onlyDelegated() {
+        require(address(this) != _SELF, NotDelegated());
         _;
     }
 
-    modifier onlyWithoutPrefund(uint256 missingAccountFunds) {
-        if (missingAccountFunds != 0) {
-            revert OnlyWithoutPrefund();
-        }
+    modifier onlyEntryPoint() {
+        require(msg.sender == ENTRY_POINT, UnsupportedEntryPoint());
         _;
     }
 
+    receive() external payable onlyDelegated {}
+
     function validateUserOp(PackedUserOperation calldata userOp, bytes32 userOpHash, uint256 missingAccountFunds)
         external
-        view
         onlyEntryPoint
```

Comment:
we don't need `onlyDelegated` here - because you will never be able to authorize the `commit`